product/src/main/java/com/yas/product/controller/ProductController.java

```diff
@@ -66,8 +66,13 @@ public ResponseEntity<List<ProductThumbnailVm>> getProductsByBrand(@PathVariable
     }
 
     @GetMapping("/storefront/category/{categorySlug}/products")
-    public ResponseEntity<List<ProductThumbnailVm>> getProductsByCategory(@PathVariable String categorySlug) {
-        return ResponseEntity.ok(productService.getProductsByCategory(categorySlug));
+    public ResponseEntity<ProductListGetFromCategoryVm> getProductsByCategory(
+            @RequestParam(value = "pageNo", defaultValue = "0", required = false) int pageNo,
+            @RequestParam(value = "pageSize", defaultValue = "2", required = false) int pageSize,
+            @RequestParam(value = "product-name", defaultValue = "", required = false) String productName,
```

Comment:
Actually, I think we don't need to filter product name here.

Reply:
I'm sorry , I mean Search for product name, but I mistakenly recorded it as Filter, I fixed it.

Reply:
I mean we don't need to the search by product name in the view product by category screen. If customer know the product name, they should use the global search function.

In the view product by category. We just need to get product by category and have pagingnation. That would be enough